include/Board.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "Piece.h"
+#include "macros.h"
+#include "Factory.h"
+#include <memory>
+#include <iostream>
+
+
+
+class Board {
+public:
+	Board(const std::string& board);
+	int getMove(const std::string& move);
+
+private:
```

Comment:
avoid accessing implementation details like the m_board array directly. Instead, provide appropriate member functions to interact with the board.